src/tests/ftest/util/data_mover_test_base.py

```diff
@@ -1190,3 +1194,88 @@ def run_datamover(self, test_desc=None,
                 self.fail("stderr expected {}: {}".format(s, test_desc))
 
         return result
+
+    def run_dm_activities_with_ior(self, tool, create_dataset=False):
+        """Generic method to perform varios datamover activities
+           using ior
+        Args:
+            tool(str): specify the tool name to be used
+            create_dataset(bool): boolean to create initial set of
+                                  data using ior. Defaults to False.
+        """
+        # Set the tool to use
+        self.set_tool(tool)
+
+        if create_dataset:
+            # create initial datasets
+            pool1 = self.create_pool()
+            cont1 = self.create_cont(pool1, oclass=self.ior_cmd.dfs_oclass.value)
+
+            # update and run ior on cont1
+            self.run_ior_with_params(
+                "DAOS", self.ior_cmd.test_file.value,
+                pool1, cont1)
+
+        # create cont2
+        cont2 = self.create_cont(pool1, oclass=self.ior_cmd.dfs_oclass.value)
+
+        # perform various datamover activities
+        if tool == 'CONT_CLONE':
+            read_back_cont = self.gen_uuid()
+            self.run_datamover(
+                self.test_id + " (cont1 to cont2)",
+                "DAOS", None, pool1, cont1,
+                "DAOS", None, pool1, read_back_cont)
+            read_back_pool = self.pool[0]
+        elif tool == 'DSERIAL':
+            # Create pool2
+            pool2 = self.create_pool()
+            # Use dfuse as a shared intermediate for serialize + deserialize
+            dfuse_cont = self.create_cont(pool1, oclass=self.ior_cmd.dfs_oclass.value)
+            self.start_dfuse(self.dfuse_hosts, pool1, dfuse_cont)
+            self.serial_tmp_dir = self.dfuse.mount_dir.value
+
+            # Serialize/Deserialize cont1 to a new cont2 in pool2
+            result = self.run_datamover(
+                self.test_id + " (cont1->HDF5->cont2)",
+                "DAOS_UUID", None, pool1, cont1,
+                "DAOS_UUID", None, pool2, None)
+
+            # Get the destination cont2 uuid
+            read_back_cont = self.parse_create_cont_uuid(result.stdout_text)
+            read_back_pool = pool2
+        else:
+            # copy from daos cont1 to cont2
+            self.run_datamover(
+                self.test_id + " (cont1 to cont2)",
+                "DAOS", "/", pool1, cont1,
+                "DAOS", "/", pool1, cont2)
```

Comment:
We should change this `else` to `else if`, and have the `else` error for an unsupported tool, since there will very likely be new tools handled by the test base in the future

Reply:
Sure. Will do.

Reply:
Done